sdk/src/network-client.ts

```diff
@@ -1198,24 +1198,59 @@ class AleoNetworkClient {
    * const transaction = await networkClient.waitForTransactionConfirmation(transactionId);
    */
   async waitForTransactionConfirmation(
-      transactionId: string,
-      checkInterval: number = 2000,  // Poll every 2 seconds
-      timeout: number = 45000        // Timeout after 45 seconds
-  ): Promise<Transaction> {
+    transactionId: string,
+    checkInterval: number = 2000,
+    timeout: number = 45000
+  ): Promise<ConfirmedTransactionJSON> {
     const startTime = Date.now();
-
-    return new Promise<Transaction>((resolve, reject) => {
+  
+    return new Promise((resolve, reject) => {
       const interval = setInterval(async () => {
+        const elapsed = Date.now() - startTime;
+  
+        if (elapsed > timeout) {
+          clearInterval(interval);
+          return reject(new Error("Transaction confirmation timed out"));
```

Comment:
In general in this method we want the errors to provide helpful information about what action to take in response to the error.

```suggestion
          return reject(new Error(`Transaction ${transactionId} did not appear after the timeout period of ${interval}ms - consider resubmitting the transaction`));
```